graylog2-server/src/main/java/org/graylog2/rest/resources/cluster/ClusterLookupTableResource.java

```diff
@@ -0,0 +1,114 @@
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import org.apache.shiro.authz.annotation.RequiresAuthentication;
+import org.apache.shiro.authz.annotation.RequiresPermissions;
+import org.graylog2.audit.jersey.NoAuditEvent;
+import org.graylog2.cluster.Node;
+import org.graylog2.cluster.NodeNotFoundException;
+import org.graylog2.cluster.NodeService;
+import org.graylog2.rest.RemoteInterfaceProvider;
+import org.graylog2.rest.resources.system.RemoteLookupTableResource;
+import org.graylog2.rest.resources.system.responses.LookupTableCachePurgingNodeResponse;
+import org.graylog2.shared.rest.resources.ProxiedResource;
+import org.graylog2.shared.security.RestPermissions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import retrofit2.Response;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.validation.constraints.NotEmpty;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.concurrent.ExecutorService;
+
+/**
+ * The primary objective of this API is to provide facilities for managing Lookup Tables on the cluster level.
+ * Originally was introduced to perform cluster-wide Cache purging.
+ */
+@RequiresAuthentication
+@Api(value = "Cluster/LookupTable")
+@Path("/cluster/system/lookup")
+@Produces(MediaType.APPLICATION_JSON)
+public class ClusterLookupTableResource extends ProxiedResource {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ClusterLookupTableResource.class);
+
+    @Inject
+    public ClusterLookupTableResource(NodeService nodeService,
+                                      RemoteInterfaceProvider remoteInterfaceProvider,
+                                      @Context HttpHeaders httpHeaders,
+                                      @Named("proxiedRequestsExecutorService") ExecutorService executorService) throws NodeNotFoundException {
+        super(httpHeaders, nodeService, remoteInterfaceProvider, executorService);
+    }
+
+    /**
+     * Calls {@link org.graylog2.rest.resources.system.lookup.LookupTableResource#performPurge(String, String)}
+     * on all active nodes and returns results per node
+     */
+    @POST
+    @Timed
+    @Path("tables/{idOrName}/purge")
+    @ApiOperation(value = "Purge Lookup Table Cache on the cluster-wide level")
+    @NoAuditEvent("Cache purge only")
+    @RequiresPermissions(RestPermissions.LOOKUP_TABLES_READ)
+    public List<LookupTableCachePurgingNodeResponse> performPurge(
+            @ApiParam(name = "idOrName") @PathParam("idOrName") @NotEmpty String idOrName,
+            @ApiParam(name = "key") @QueryParam("key") String key) {
+
```

Comment:
I think this can largely simplified by using `getForAllNodes` from the `ProxiedResource`
Kind of like:
```
+        return getForAllNodes(remoteResource -> remoteResource.performPurge(idOrName, key), createRemoteInterfaceProvider(RemoteLookupTableResource.class));
```
This means the response also has to change.
I'd suggest this is good enough for our purpose:
```
    public Map<String, Optional<Boolean>> performPurge(
```
We can change the `LookupTableResource` endpoint to return a `boolean` instead of a `void`

Reply:
That was the first thing I had tried, but decided to retreat because a) it entails an API change b) `getForAllNodes` yields the same result for `No Content` responses and for failed requests.

Reply:
On top of that with this custom approach we provide more fine-grained error messages.

Reply:
for a) it should be fine if we go from `no content` to returning a response.

I agree that it's not as fine-grained, but still good enough.
It also uses a established response for cluster api calls.

Reply:
As agreed here is a generic solution for future generations: 2633ea14194847156939d7f1fb3381cb586364d6